transports/websocket/src/framed.rs

```diff
@@ -0,0 +1,431 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use bytes::BytesMut;
+use crate::{error::Error, tls};
+use futures::{future::{self, Either, Loop}, prelude::*, try_ready};
+use libp2p_core::{
+    Transport,
+    either::EitherOutput,
+    multiaddr::{Protocol, Multiaddr},
+    transport::{ListenerEvent, TransportError}
+};
+use log::{debug, trace};
+use tokio_rustls::{client, server};
+use soketto::{base, connection::{Connection, Mode}, handshake::{self, Redirect, Response}};
+use std::io;
+use tokio_codec::{Framed, FramedParts};
+use tokio_io::{AsyncRead, AsyncWrite};
+use tokio_rustls::webpki;
+use url::Url;
+
+/// Max. number of payload bytes of a single frame.
+const MAX_DATA_SIZE: u64 = 256 * 1024 * 1024;
+
+/// A Websocket transport whose output type is a [`Stream`] and [`Sink`] of
+/// frame payloads which does not implement [`AsyncRead`] or
+/// [`AsyncWrite`]. See [`crate::WsConfig`] if you require the latter.
+#[derive(Debug, Clone)]
+pub struct WsConfig<T> {
```

Comment:
What about `FramedWsConfig`? So that you actually know that this is not the regular `WsConfig`.

Reply:
I do not like the repetition on namespaces or module names in identifiers. Instead I prefer to import the module and refer to those identifiers with the module prefix, i.e. `io::Error` instead of `IoError`, `framed::WsConfig` instead of `FramedWsConfig`, `tls::Config` instead of `TlsConfig`. Otherwise names are just blown up for no good reason.

Reply:
It's not so much a problem of naming  as it is a problem of namespaces. I prefer having one `websocket` namespace with `websocket::TlsConfig`, `websocket::FramedConfig`, etc. rather than `framed::Config` or `tls::Config` which don't indicate much.

Reply:
This naming scheme reintroduces namespaces as part of identifier names.
`websocket::framed::Config` and `websocket::tls::Config` use the language provided mechanism for name spacing, whereas `websocket::FramedConfig` and `websocket::TlsConfig` do not.